hapi-loader/src/main/java/com/arkhn/hapi/loader/ResourceConsumer.java

```diff
@@ -0,0 +1,78 @@
+package com.arkhn.hapi.loader;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.kafka.annotation.KafkaHandler;
+import org.springframework.kafka.annotation.KafkaListener;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.api.dao.DaoRegistry;
+import ca.uhn.fhir.jpa.api.dao.IFhirResourceDao;
+import ca.uhn.fhir.parser.IParser;
+
+@SpringBootApplication
+public class ResourceConsumer {
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SpringApplication.run(ResourceConsumer.class, args);
+        } catch (Exception e) {
+            System.exit(1);
+        }
+    }
+
+    @Bean
+    public ResourceListener resourceListener() {
+        return new ResourceListener();
+    }
+
+    @KafkaListener(id = "resource-loader", topics = "${hapi.loader.kafka.topic}", containerFactory = "kafkaListenerContainerFactory", autoStartup = "true", concurrency = "${hapi.loader.concurrency}")
+    public static class ResourceListener {
+
+        @Autowired
+        DaoRegistry daoRegistry;
+
+        @Autowired
+        FhirContext myFhirContext;
+
+        @KafkaHandler
+        public void listen(String message) {
+            IParser parser = myFhirContext.newJsonParser();
+            IBaseResource r = parser.parseResource(message);
+
+            @SuppressWarnings("unchecked")
+            IFhirResourceDao<IBaseResource> dao = daoRegistry.getResourceDao(r.getClass().getSimpleName());
+
+            dao.update(r);
+
+            // TODO: error handling
+
+            // TODO: produce "load" events
+
+            // // THE FOLLOWING CODE IS THE "BATCH UPDATE" VERSION
```

Comment:
Can you tell that we don't do the batch version right now for performances reasons?

Reply:
done